vibrato/src/tokenizer.rs

```diff
@@ -84,7 +85,10 @@ impl Tokenizer {
         Worker::new(self)
     }
 
-    pub(crate) fn build_lattice(&self, sent: &Sentence, lattice: &mut Lattice) {
+    pub(crate) fn build_lattice<C>(&self, sent: &Sentence, lattice: &mut Lattice, connector: &C)
+    where
+        C: ConnectorCost,
```

Comment:
The argument `connector` must be guaranteed to be `self.dict.connector`. We know that this `build_lattice` is used only in `Worker::tokenize`, but as long as the accessibility is `pub(crate)`, there is a risk that it will not be guaranteed.

How about modifying it as follows?

```rust
    pub(crate) fn build_lattice(&self, sent: &Sentence, lattice: &mut Lattice) {
        match self.dict.connector() {
            ConnectorWrapper::Matrix(c) => self.build_lattice_inner(sent, lattice, c),
        }
    }

    fn build_lattice_inner<C>(&self, sent: &Sentence, lattice: &mut Lattice, connector: &C)
    where
        C: ConnectorCost,
    {
```